File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
# 에러토스테네스의 체
def is_prime_num(a):
    arr = [1] * (a + 1)
    arr[0] = 0
    arr[1] = 0

    for i in range(2, a + 1):
        if arr[i] == 1:
            j = 2

            while (i * j) <= a:
                arr[i * j] = 0
                j += 1

    return arr

n = int(input())
num = list(map(int, input().split()))

arr = is_prime_num(1000)
cnt = 0

for i in num:
    if arr[i] == 1:
        cnt += 1

print(cnt)

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.cc
#include <iostream>
#include <vector>

using namespace std;

// 에라토스테네스의 체
vector<bool> is_prime_num(int a) {
  vector<bool> arr(a + 1, true);
  arr[0] = false;
  arr[1] = false;

  for (int i = 2; i * i <= a; i++) {
    if (arr[i]) {
      for (int j = i * i; j <= a; j += i) {
        arr[j] = false;
        }
    }
  }
  return arr;
}

int main() {
  int n;
  int cnt = 0;
  vector<bool> primes = is_prime_num(1001);

  cin >> n;

  for (int i = 0; i < n; i++) {
    int num;
    cin >> num;

    if (primes[num]) {
      cnt++;
    }
  }

  cout  << cnt;

  return 0;
}